### PR TITLE
Re-enable public api check

### LINKED ISSUE #2526: Reenable public API check
Was disabled temporarily while Metric was in very early stage. We need to re-enable it and document the public API for metrics
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/build/Common.prod.props#L11-L15

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #2526 

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry/.publicApi/netstandard2.0/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -1,8 +1,145 @@
...
+OpenTelemetry.Metrics.ExportModes.Pull = 2 -> OpenTelemetry.Metrics.ExportModes
+OpenTelemetry.Metrics.ExportModes.Push = 1 -> OpenTelemetry.Metrics.ExportModes
+OpenTelemetry.Metrics.ExportModesAttribute
+OpenTelemetry.Metrics.ExportModesAttribute.ExportModesAttribute(OpenTelemetry.Metrics.ExportModes supported) -> void
+OpenTelemetry.Metrics.ExportModesAttribute.Supported.get -> OpenTelemetry.Metrics.ExportModes
+OpenTelemetry.Metrics.HistogramConfiguration
+OpenTelemetry.Metrics.HistogramConfiguration.BucketBounds.get -> double[]
+OpenTelemetry.Metrics.HistogramConfiguration.BucketBounds.set -> void
+OpenTelemetry.Metrics.HistogramConfiguration.HistogramConfiguration() -> void
+OpenTelemetry.Metrics.IDataPoint
```
Comment (cijothomas) [inline src/OpenTelemetry/.publicApi/netstandard2.0/PublicAPI.Unshipped.txt:45]: could be cleaned up now.
(will be a separate PR along with reviewing entire public API)

**Review by cijothomas — APPROVED** [review]
